dotCMS/src/main/java/com/dotmarketing/common/db/DotDatabaseMetaData.java

```diff
@@ -0,0 +1,313 @@
+
+                if (!foreignKey.getForeignKeyColumnNames().contains(foreignKeyColumnName)) {
+                    return false;
+                }
+            }
+
+            allMatch = true;
+        }
+
+        return allMatch;
+    }
+
+    /**
+     * Reads all the foreign keys associated to every table in the specified
+     * list.
+     *
+     * @param tables
+     *            - The list of database tables whose foreign keys will be
+     *            retrieved.
+     *
+     * @return The list of foreign keys associated to the tables.
+     */
+
+    public List<ForeignKey> getForeignKeys(final String... tables) {
+
+        return getForeignKeys(DbConnectionFactory.getConnection(), tables);
+    }
+
+    /**
+     * Reads all the foreign keys associated to every table in the specified
+     * list.
+     *
+     * @param conn
+     *            - The database connection object to access the dotCMS data.
+     * @param tables
+     *            - The list of database tables whose foreign keys will be
+     *            retrieved.
+     *
+     * @return The list of foreign keys associated to the tables.
+     */
+    public List<ForeignKey> getForeignKeys(final Connection conn, final String... tables) {
+
+        final List<ForeignKey> foreignKeys = new ArrayList<>();
+        final Map<String, ForeignKey> foreignKeyMap = new HashMap<>();
+        DatabaseMetaData databaseMetaData     = null;
+        ResultSet        resultSet            = null;
+        ForeignKey       foreignKey           = null;
+        String           schema               = null;
+
+        try {
+
+            databaseMetaData = conn.getMetaData();
+
+            for (String table : tables) {
+
+                if (DbConnectionFactory.isOracle()) {
+                    table = table.toUpperCase();
+                    schema = databaseMetaData.getUserName();
+                }
+
+                resultSet = databaseMetaData.getImportedKeys
+                        (conn.getCatalog(), schema, table);
+
+                // Iterates over the foreign foreignKey columns
+                while (resultSet.next()) {
+
+                    foreignKey = new ForeignKey(resultSet.getString("PKTABLE_NAME"),
+                            resultSet.getString("FKTABLE_NAME"),
+                            resultSet.getString("FK_NAME")
+                            );
+
+                    final String pkColumnName = resultSet.getString("PKCOLUMN_NAME");
+                    final String fkColumnName = resultSet.getString("FKCOLUMN_NAME");
+
+                    foreignKey.addPrimaryColumnName(pkColumnName);
+                    foreignKey.addForeignColumnName(fkColumnName);
+
+                    if (!foreignKeys.contains(foreignKey)) {
+                        foreignKeys.add(foreignKey);
+                        foreignKeyMap.put(foreignKey.fkName(), foreignKey);
+                    } else {
+                        if (DbConnectionFactory.isMsSql()) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)